src/XamlX/Parsers/XDocumentXamlParser.cs

```diff
@@ -126,8 +131,29 @@ static IXamlAstValueNode ParseTextValueOrMarkupExtension(string ext, XElement xe
                         try
                         {
 
-                            return SystemXamlMarkupExtensionParser.SystemXamlMarkupExtensionParser.Parse(info, ext,
+                            var extensionObject = SystemXamlMarkupExtensionParser.SystemXamlMarkupExtensionParser.Parse(info, ext,
                                 t => ParseTypeName(info, t, xel));
+
+                            if (extensionObject is XamlAstObjectNode { Type: XamlAstXmlTypeReference xmlType } astObject)
```

Comment:
1) I believe that has to be done recursively
2) It's a wrong place to handle x:TypeArguments, you should convert namespace-prefixed properties to XAML directives, see https://github.com/kekekeks/XamlX/blob/a148b3ae0eaa114d2d61518ebab14732a9b17833/src/XamlX/Parsers/XDocumentXamlParser.cs#L199-L205

Reply:
@kekekeks done